tests/test_snippets/test_parsons.py

```diff
@@ -0,0 +1,88 @@
+import os
+import hedy
+from website.yaml_file import YamlFile
+import unittest
+from tests.Tester import HedyTester, Snippet
+from parameterized import parameterized
+from hedy_content import ALL_KEYWORD_LANGUAGES
+
+# Set the current directory to the root Hedy folder
+os.chdir(os.path.join(os.getcwd(), __file__.replace(os.path.basename(__file__), '')))
+
+unique_snippets_table = set()
+
+def collect_snippets(path):
+    Hedy_snippets = []
+    files = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f)) and f.endswith('.yaml')]
+    for file in files:
+        lang = file.split(".")[0]
+        file = os.path.join(path, file)
+        yaml = YamlFile.for_file(file)
+        levels = yaml.get('levels')
+
+        for level, content in levels.items():
+            level_number = int(level)
+            if level_number > hedy.HEDY_MAX_LEVEL:
+                print('content above max level!')
+            else:
+                try:
+                    for exercise_id, exercise in levels[level].items():
+                        lines = exercise.get('code_lines')
+                        code = ""
+                        # The lines have a letter: A: ..., B:...., C:....
+                        for letter in "ABCDEFGHIJKLMNOPQRSTUVWXYZ":
+                            line = lines.get(letter)
+                            if line:
+                                code += line + "\n"
+                            else:
+                                break
+                        if hash(code) in unique_snippets_table:
+                            print("Identical code already being tested...")
+                            continue
+                        else:
+                            unique_snippets_table.add(hash(code))
+                        Hedy_snippets.append(Snippet(filename=file, level=level, field_name=f"{exercise_id}", code=code))
+                except:
+                    print(f'Problem reading commands yaml for {lang} level {level}')
+
+
+    return Hedy_snippets
+
+def translate_keywords_in_snippets(snippets):
```

Comment:
I know I caused the existing duplication cause I was lazy but can we put this in one place (in Tester?)